spring-cloud-deployer-cloudfoundry/src/main/java/org/springframework/cloud/deployer/spi/cloudfoundry/CloudFoundryAppDeployer.java

```diff
@@ -212,7 +213,14 @@ public void undeploy(String id) {
 
 	@Override
 	public String getLog(String id) {
-		return applicationLogAccessor.getLog(id, Duration.ofSeconds(this.deploymentProperties.getApiTimeout()));
+		AppStatus status = status(id);
```

Comment:
Wdyt about using streams API to do this, something like:
```java
@Override
public String getLog(String id) {
  AppStatus status = status(id);
  String cfGuid = status.getInstances().values().stream()
		  .map((appInstanceStatus) -> appInstanceStatus.getAttributes().get("cf-guid"))
		  .filter(Objects::nonNull)
		  .findFirst()
		  .orElseThrow(() -> new IllegalArgumentException("Unable to find cf-guid"));
  return applicationLogAccessor.getLog(cfGuid, Duration.ofSeconds(this.deploymentProperties.getApiTimeout()));
}
```
Also, it currently does not handle the case where the guid was not found (see the task code below)